.github/workflows/cpp.yaml

```diff
@@ -0,0 +1,47 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# This is a basic workflow to help you get started with Actions
+
+name: C++
+
+# Controls when the action will run. Triggers the workflow on push or pull request
+# events but only for the master branch
+on:
+  push:
+    branches:
+      - master
+      - dev
+  pull_request:
+    branches:
+      - master
+      - dev
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  build-ubuntu-cpp:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v1
+      - name: install-build-dependencies
+        run: sudo apt-get install -y libgoogle-glog-dev libprotobuf-dev protobuf-compiler libncurses-dev libopenblas-dev gfortran libblas-dev liblapack-dev libatlas-base-dev swig libcurl3-dev cmake dh-autoreconf  
+      - name: configure
+        run: mkdir build && cd build && cmake -DUSE_PYTHON=NO ..
```

Comment:
Since we are using DNNL as one of the C++ computational kernal (other example of comp kernals are BLAS etc.)

The cmake flag should include -DUSE_DNNL=ON

FYI:

1. This is the cmake command in conda version of SINGA
https://github.com/apache/singa/blob/dev/tool/conda/singa/build.sh#L43
```
cmake -DCMAKE_INSTALL_PREFIX=$PREFIX -DUSE_CUDA=$USE_CUDA \
	-DUSE_PYTHON3=ON -DUSE_DNNL=ON -DUSE_DIST=$USE_DIST -DCMAKE_OSX_SYSROOT=${CONDA_BUILD_SYSROOT} ..
```


2. This is the cmake command in docker version of SINGA
https://github.com/apache/singa/blob/dev/tool/docker/devel/ubuntu/cpu/Dockerfile#L67
`cmake -DENABLE_TEST=ON -DUSE_PYTHON3=ON -DUSE_DNNL=ON ..`

You may consider refer to both versions on how to include the DNNL library

Thanks!

Reply:
Thank you for the revision, here are some comments:

1. The MKL-DNN is now called [oneDNN](https://github.com/oneapi-src/oneDNN)

2. If the flag -DUSE_DNNL is always required to pass the C++ tests, then why it is an optional flag? It should be always ON by default since the system does not work without it. (i.e. the system does not pass the unit tests without it)

3. If the C++ unit tests can work without the -DUSE_DNNL flag except few unit tests, then I propose to add "ifdef USE_DNNL" around the unit tests that require DNNL:

```
#ifdef USE_DNNL

.. unit test code

#endif
```


So that these unit tests are not compiled if they don't support the given configuration to cmake.

What do you think?

Reply:
Thanks so much for your ideas. I will think about your solutions 2 and 3, and discuss with others.

I think the DNNL (oneDNN) is very important in the test. Is the Github Actions code able to provide the library for test? We are using DNNL version 1.1 as indicated in https://github.com/apache/singa/blob/dev/tool/docker/devel/ubuntu/cpu/Dockerfile#L52

Reply:
Yes, any shell script for build can be used. I added the [oneDNN](https://github.com/apache/singa/pull/761/checks?check_run_id=871284806) check and it works fine.